packages/sanity/src/_internal/cli/actions/schema/storeSchemasAction.ts

```diff
@@ -0,0 +1,136 @@
+import {readFileSync} from 'node:fs'
+import path, {join, resolve} from 'node:path'
+
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import chalk from 'chalk'
+
+import {
+  type CreateManifest,
+  type ManifestSchemaType,
+  type ManifestWorkspaceFile,
+} from '../../../manifest/manifestTypes'
+import {MANIFEST_FILENAME} from '../manifest/extractManifestAction'
+import {SANITY_WORKSPACE_SCHEMA_ID} from './schemaListAction'
+
+export interface StoreManifestSchemasFlags {
+  'path'?: string
+  'workspace'?: string
+  'id-prefix'?: string
+  'schema-required'?: boolean
+  'verbose'?: boolean
+}
+
+export default async function storeSchemasAction(
+  args: CliCommandArguments<StoreManifestSchemasFlags>,
+  context: CliCommandContext,
+): Promise<Error | undefined> {
+  const flags = args.extOptions
+  if (typeof flags.path === 'boolean') throw new Error('Path is empty')
+  if (typeof flags['id-prefix'] === 'boolean') throw new Error('Id prefix is empty')
+  if (typeof flags.workspace === 'boolean') throw new Error('Workspace is empty')
+
+  const schemaRequired = flags['schema-required']
+  const workspaceName = flags.workspace
+  const idPrefix = flags['id-prefix']
```

Comment:
Could probably move these up over the checks, and use the variables for validation instead? Then you coul also spread them out of flags, and the validations would type-narrow in the below code.